### PR TITLE
[FLINK-6731] [tests] Activate strict checkstyle for flink-tests

### PR DESCRIPTION
**Pull request by greghogan**

### CODE REVIEW

**Comment by zentol** [comment]
The `pom.xml` wasn't updated to run the checkstyle plugin.

**Comment by zentol** [comment]
Import changes look ok.

**Comment by zentol** [comment]
Whitespace changes look ok.

File: flink-tests/src/test/java/org/apache/flink/test/windowing/sessionwindows/GeneratorConfiguration.java [hunk 1]
```diff
@@ -36,9 +36,9 @@
 	private final long maxAdditionalSessionGap;
 
 	public GeneratorConfiguration(long allowedLateness,
```
Comment (zentol) [inline flink-tests/src/test/java/org/apache/flink/test/windowing/sessionwindows/GeneratorConfiguration.java]: let's move the `allowedLateness` argument to the next line as well.

File: flink-tests/src/test/java/org/apache/flink/test/windowing/sessionwindows/GeneratorEventFactory.java [hunk 2]
```diff
@@ -39,9 +39,9 @@
 	 * @return event for an keyed event generator
 	 */
 	E createEvent(K key,
```
Comment (zentol) [inline flink-tests/src/test/java/org/apache/flink/test/windowing/sessionwindows/GeneratorEventFactory.java]: move to next line

File: flink-tests/src/test/java/org/apache/flink/test/accumulators/AccumulatorIterativeITCase.java [hunk 3]
```diff
@@ -30,6 +29,9 @@
 
 import org.junit.Assert;
 
+/**
+ *
```
Comment (greghogan) [inline flink-tests/src/test/java/org/apache/flink/test/accumulators/AccumulatorIterativeITCase.java]: Updated.

File: flink-tests/src/test/java/org/apache/flink/test/checkpointing/RescalingITCase.java [hunk 4]
```diff
@@ -1035,4 +1040,43 @@ public void restoreState(Integer state) throws Exception {
 			counterPartitions.add(state);
 		}
 	}
+
+	/**
+	 * Copied from io.netty.util.internal.ConcurrentSet.
+	 *
+	 * @param <E> the type of elements maintained by this set
+	 */
+	private static final class ConcurrentSet<E> extends AbstractSet<E> implements Serializable {
```
Comment (greghogan) [inline flink-tests/src/test/java/org/apache/flink/test/checkpointing/RescalingITCase.java]: This dependency was removed in a recent commit so am deleting the class.

File: flink-tests/src/test/java/org/apache/flink/test/checkpointing/StreamCheckpointNotifierITCase.java [hunk 5]
```diff
@@ -324,7 +322,7 @@ public void notifyCheckpointComplete(long checkpointId) {
 	private static class LeftIdentityCoRichFlatMapFunction extends RichCoFlatMapFunction<Long, Long, Long>
 			implements CheckpointListener {
 
-		static final List<Long>[] completedCheckpoints = createCheckpointLists(PARALLELISM);
```
Comment (greghogan) [inline flink-tests/src/test/java/org/apache/flink/test/checkpointing/StreamCheckpointNotifierITCase.java]: I've fixed the instances in this file. Variables are now uppercase.

File: flink-tests/src/test/java/org/apache/flink/test/misc/SuccessAfterNetworkBuffersFailureITCase.java [hunk 6]
```diff
@@ -168,6 +173,6 @@ private static void runKMeans(ExecutionEnvironment env) throws Exception {
 
 		clusteredPoints.output(new DiscardingOutputFormat<Tuple2<Integer, KMeans.Point>>());
 
-		env.execute("KMeans Example");
+		env.execute("kmeans Example");
```
Comment (greghogan) [inline flink-tests/src/test/java/org/apache/flink/test/misc/SuccessAfterNetworkBuffersFailureITCase.java]: Agreed.

File: flink-tests/src/test/java/org/apache/flink/test/windowing/sessionwindows/GeneratorConfiguration.java (hunk 1 above)
Comment (greghogan) [inline flink-tests/src/test/java/org/apache/flink/test/windowing/sessionwindows/GeneratorConfiguration.java]: Done.

File: flink-tests/src/test/java/org/apache/flink/test/windowing/sessionwindows/GeneratorEventFactory.java (hunk 2 above)
Comment (greghogan) [inline flink-tests/src/test/java/org/apache/flink/test/windowing/sessionwindows/GeneratorEventFactory.java]: Done.

File: flink-tests/src/test/java/org/apache/flink/test/iterative/KMeansWithBroadcastSetITCase.java [hunk 7]
```diff
@@ -65,7 +68,7 @@ public Centroid map(String c) {
 					}
 				});
 
-		// set number of bulk iterations for KMeans algorithm
+		// set number of bulk iterations for kmeans algorithm
```
Comment (zentol) [inline flink-tests/src/test/java/org/apache/flink/test/iterative/KMeansWithBroadcastSetITCase.java]: Another instance of "kmeans" renaming

File: flink-tests/src/test/java/org/apache/flink/test/classloading/jar/KMeansForTest.java [hunk 8]
```diff
@@ -70,7 +69,7 @@ public static void main(String[] args) throws Exception {
 		DataSet<Centroid> centroids = env.fromElements(centersData.split("\n"))
 				.map(new TupleCentroidConverter());
 
-		// set number of bulk iterations for KMeans algorithm
+		// set number of bulk iterations for kmeans algorithm
```
Comment (zentol) [inline flink-tests/src/test/java/org/apache/flink/test/classloading/jar/KMeansForTest.java]: and another

**Comment by zentol** [comment]
+1 with green light from travis.

File: flink-tests/src/test/java/org/apache/flink/test/iterative/KMeansWithBroadcastSetITCase.java (hunk 7 above)
Comment (greghogan) [inline flink-tests/src/test/java/org/apache/flink/test/iterative/KMeansWithBroadcastSetITCase.java]: I reverted these and the additional instances. It looks to have happened when renaming `KMeansSingleStepTest#KMeans`.

**Pull request closed without merging**